src/Microsoft.Data.Sqlite/SqliteCommand.cs

```diff
@@ -154,9 +162,14 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
                     throw new InvalidOperationException(Strings.FormatMissingParameters(string.Join(", ", unboundParams)));
                 }
 
-                rc = NativeMethods.sqlite3_step(stmt);
                 try
                 {
+                    rc = NativeMethods.sqlite3_step_blocking(Connection.DbHandle, stmt, CommandTimeout*1000);
```

Comment:
This is cleaner than I thought it would be. Only one command ever runs on the connection at a time, so the fact that the timeout is actually associated with the connection doesn't ever really mater.